app.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @peoples = []
+  end
+
+  def list_all_books
+    @books.each { |book| puts "Title:- #{book.title}   Author:- #{book.author} \n" }
+  end
+
+  def list_all_peoples
+    @peoples.each do |people|
+      puts "[#{people.class.name}] Name:- #{people.name} ID:- #{people.id} Age:- #{people.age} \n"
+    end
+  end
+
+  def create_teacher(age, specialization, name, parent_permission)
+    @peoples << Teacher.new(age, specialization, name: name, parent_permission: parent_permission)
+  end
+
+  def create_student(age, name, parent_permission)
```

Comment:
- [ ] Attempts to create a student or teacher keep failing with an ArgumentError. kindly fix this to ensure that your application works smoothly. 

> Screenshot
<img width="712" alt="Screenshot 2023-05-20 at 12 25 21" src="https://github.com/Koshao-sys/OOP-school-library/assets/71140133/60267a4e-132d-492c-aa52-4ce1bc762f97">